rs/rust_canisters/call_loop_canister/src/main.rs

```diff
@@ -0,0 +1,63 @@
+use candid::candid_method;
+use ic_cdk::call::Call;
+use ic_cdk_macros::update;
+use ic_principal::Principal;
+use std::future::IntoFuture;
+
+/// Takes the total number of bytes to send in a single message (in megabytes).
+#[candid_method(update)]
+#[update]
+async fn send_calls(megabytes_to_send: u32) {
+    let calls = (0..megabytes_to_send)
+        .map(|i| {
+            let mut slice = [0; 29];
+            slice[..4].copy_from_slice(&i.to_le_bytes());
+            let canister = Principal::from_slice(&slice);
+            Call::unbounded_wait(canister, "")
+                .with_raw_args(&[5; 1024 * 1024])
+                .into_future()
+        })
+        .collect::<Vec<_>>();
+    let _ = futures::future::join_all(calls).await;
+}
+
+// When run on native this prints the candid service definition of this
+// canister, from the methods annotated with `candid_method` above.
+//
+// Note that `cargo test` calls `main`, and `export_service` (which defines
+// `__export_service` in the current scope) needs to be called exactly once. So
+// in addition to `not(target_family = "wasm")` we have a `not(test)` guard here
+// to avoid calling `export_service`, which we need to call in the test below.
```

Comment:
Not sure what we get exactly from all of this. Is there some actual scenario when main would run natively? I only see the state machine test which runs the canister of course.

Reply:
It gets called when generating the `did` file and also in the test which asserts the `did` file is correct.